### PR TITLE
Remove unnecessary parentheses from `JdbcDecoder`, `JdbcEncoder` and `Setter` summoners

### PR DESCRIPTION
**Pull request by guizmaii**
Needed for https://github.com/oyvindberg/typo/pull/57

### CODE REVIEW

File: core/src/main/scala/zio/jdbc/JdbcEncoder.scala [hunk 1]
```diff
@@ -29,8 +29,7 @@ trait JdbcEncoder[-A] {
 }
 
 object JdbcEncoder extends JdbcEncoder0LowPriorityImplicits {
-  def apply[A]()(implicit encoder: JdbcEncoder[A]): JdbcEncoder[A] = encoder
```
Comment (guizmaii) [inline core/src/main/scala/zio/jdbc/JdbcEncoder.scala:32]: There's no need for these `()` here

File: core/src/test/scala/zio/jdbc/TestConnection.scala [hunk 2]
```diff
@@ -166,15 +167,15 @@ class DummyPreparedStatement(failNext: Boolean, elemns: Int) extends PreparedSta
 
   var closed = false
 
-  override def unwrap[T <: Object](iface: Class[T]) = ???
```
Comment (guizmaii) [inline core/src/test/scala/zio/jdbc/TestConnection.scala:169]: It was not compiling in Scala 2.12 with this `<: Object`

**Review by adamgfraser — CHANGES_REQUESTED** [review]
There are a lot of extraneous changes here. Can we focus on just the specific issue?

**Comment by guizmaii** [comment]
Superseded by https://github.com/zio/zio-jdbc/pull/167

**Pull request closed without merging**